src/wwwroot/js/genpage/main.js

```diff
@@ -609,6 +628,51 @@ function toggleStar(path, rawSrc) {
 
 defaultButtonChoices = 'Use As Init,Edit Image,Star,Reuse Parameters';
 
+function getFullsrc(src) {
+    let fullSrc = src;
+    if (fullSrc.startsWith("http://") || fullSrc.startsWith("https://")) {
+        fullSrc = fullSrc.substring(fullSrc.indexOf('/', fullSrc.indexOf('/') + 2));
+    }
+    if (fullSrc.startsWith('/')) {
+        fullSrc = fullSrc.substring(1);
+    }
+    if (fullSrc.startsWith('Output/')) {
+        fullSrc = fullSrc.substring('Output/'.length);
+    }
+    if (fullSrc.startsWith('View/')) {
+        fullSrc = fullSrc.substring('View/'.length);
+        let firstSlash = fullSrc.indexOf('/');
+        if (firstSlash != -1) {
+            fullSrc = fullSrc.substring(firstSlash + 1);
+        }
+    }
+    return fullSrc;
+}
+
+function getButtonsChoice() {
+    let buttonsChoice = getUserSetting('ButtonsUnderMainImages', '');
+    if (buttonsChoice == '')
+    {
+        buttonsChoice = defaultButtonChoices;
+    }
+    buttonsChoice = buttonsChoice.toLowerCase().replaceAll(' ', '').split(',');
+    return buttonsChoice;
+}
+
+function includeButton(buttons, subButtons, name, action, extraClass = '', title = '') {
```

Comment:
this should not be separated, it's unique to setCurrentImage (buttonsChoice split)